Mathlib/Data/Real/CauSeqCompletion.lean

```diff
@@ -0,0 +1,464 @@
+#align cau_seq.completion.cau_seq_zero_ne_one CauSeq.Completion.cau_seq_zero_ne_one
+
+theorem zero_ne_one : (0 : (Cauchy abv)) ≠ 1 := fun h => cau_seq_zero_ne_one <| mk_eq.1 h
+#align cau_seq.completion.zero_ne_one CauSeq.Completion.zero_ne_one
+
+protected theorem inv_mul_cancel {x : (Cauchy abv)} : x ≠ 0 → x⁻¹ * x = 1 :=
+  Quotient.inductionOn x fun f hf => by
+    simp at hf; simp [hf]
+    exact Quotient.sound (CauSeq.inv_mul_cancel hf)
+#align cau_seq.completion.inv_mul_cancel CauSeq.Completion.inv_mul_cancel
+
+protected theorem mul_inv_cancel {x : (Cauchy abv)} : x ≠ 0 → x * x⁻¹ = 1 :=
+  Quotient.inductionOn x fun f hf => by
+    simp at hf; simp [hf]
+    exact Quotient.sound (CauSeq.mul_inv_cancel hf)
+#align cau_seq.completion.mul_inv_cancel CauSeq.Completion.mul_inv_cancel
+
+theorem ofRat_inv (x : β) : ofRat x⁻¹ = ((ofRat x)⁻¹ : (Cauchy abv)) :=
+  congr_arg mk <| by split_ifs with h <;> [simp [const_limZero.1 h], rfl]
+#align cau_seq.completion.of_rat_inv CauSeq.Completion.ofRat_inv
+
+/- porting note: This takes a long time to compile.
+   Also needed to rewrite the proof of ratCast_mk due to simp issues -/
+/-- The Cauchy completion forms a division ring. -/
+noncomputable instance Cauchy.divisionRing : DivisionRing (Cauchy abv) where
+    exists_pair_ne := ⟨0, 1, zero_ne_one⟩
+    inv_zero := inv_zero
+    mul_inv_cancel := fun x => CauSeq.Completion.mul_inv_cancel
+    ratCast := fun q => ofRat q
+    ratCast_mk := fun n d hd hnd => by rw [← ofRat_ratCast, Rat.cast_mk', ofRat_mul, ofRat_inv]; rfl
+
+theorem ofRat_div (x y : β) : ofRat (x / y) = (ofRat x / ofRat y : Cauchy abv) := by
+  simp only [div_eq_mul_inv, ofRat_inv, ofRat_mul]
+#align cau_seq.completion.of_rat_div CauSeq.Completion.ofRat_div
+
+/-- Show the first 10 items of a representative of this equivalence class of cauchy sequences.
+
+The representative chosen is the one passed in the VM to `quot.mk`, so two cauchy sequences
+converging to the same number may be printed differently.
+-/
+unsafe instance [Repr β] : Repr (Cauchy abv) where
+  reprPrec r _ :=
+    let N := 10
+    let seq := r.unquot
+    "(sorry /- " ++ Std.Format.joinSep ((List.range N).map <| repr ∘ seq) ", " ++ ", ... -/)"
+
+end
+
+section
+
+variable {α : Type _} [LinearOrderedField α]
+
+variable {β : Type _} [Field β] {abv : β → α} [IsAbsoluteValue abv]
+
+/-- The Cauchy completion forms a field. -/
+noncomputable instance Cauchy.field : Field (Cauchy abv) :=
+  { Cauchy.divisionRing, Cauchy.commRing with }
+
+end
+
+end CauSeq.Completion
+
+variable {α : Type _} [LinearOrderedField α]
+
+namespace CauSeq
+
+section
+
+variable (β : Type _) [Ring β] (abv : β → α) [IsAbsoluteValue abv]
+
+/-- A class stating that a ring with an absolute value is complete, i.e. every Cauchy
+sequence has a limit. -/
+class IsComplete : Prop where
+  /-- Every Cauchy sequence has a limit. -/
+  is_complete : ∀ s : CauSeq β abv, ∃ b : β, s ≈ const abv b
+#align cau_seq.is_complete CauSeq.IsComplete
```

Comment:
This is verbatim from the original PR, but is wrong; either the field should be called `IsComplete`, or the align statement should be `is_complete`.

Which is preferred?

Reply:
I think the field should be calles `isComplete`

Reply:
Do we need two `#align` statements? One for the class and one for the field?